### PR TITLE
Cleanup player components & name decorations

### PR DESCRIPTION
**Pull request by Pablete1234**
This is mainly a refactor of the player component. 
Simplifies the many different methods for rendering into a single method, also prefix components now appear everywhere instead of specifically chat messages and join/leave messages.
It also flattens the path to viewer-dependant name rendering

### CODE REVIEW

**Review by applenick — DISMISSED** [review]

File: core/src/main/java/tc/oc/pgm/namedecorations/NameDecorationRegistryImpl.java [hunk 1]
```diff
@@ -1,98 +1,148 @@
...
+    return decorationCache.getUnchecked(uuid).prefix;
   }
 
   public String getSuffix(UUID uuid) {
-    return provider != null ? provider.getSuffix(uuid) : "";
+    return decorationCache.getUnchecked(uuid).suffix;
+  }
+
+  public TextColor getColor(UUID uuid) {
+    MatchPlayer pl = PGM.get().getMatchManager().getPlayer(uuid);
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/namedecorations/NameDecorationRegistryImpl.java]: `player` ? :)

**Review by Electroid — APPROVED** [review]

**Comment by Electroid** [comment]
@Pablete1234 Out of draft?

**Comment by Pablete1234** [comment]
I wasn't sure if i wanted to keep on adding stuff to this, but i think it may be better to get it out of draft, merge it, and add more things to it later in another pr

**Review by Electroid — APPROVED** [review]
Refactor looks great!

**Pull request merged**